Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1

```diff
@@ -0,0 +1,795 @@
+            Write-Host $([Constants]::DoubleDashLine)	
+            return
+        }
+    }
+    else
+    {
+        Write-Host "[Step 1 of 4] Validate the user" 
+        Write-Host $([Constants]::SingleDashLine)
+    }  
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        
+        Write-Host "Connecting to Azure account..."
+        Write-Host $([Constants]::SingleDashLine)
+        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
+        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    else
+    {
+        # Setting up context for the current Subscription.
+        $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    }
+
+    Write-Host "Subscription Name: [$($context.Subscription.Name)]"
+    Write-Host "Subscription ID: [$($context.Subscription.SubscriptionId)]"
+    Write-Host "Account Name: [$($context.Account.Id)]"
+    Write-Host "Account Type: [$($context.Account.Type)]"
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Note about the required access required for remediation
+
+    Write-Host "To Configured WAF Policy for all Front Door CDN Endpoint(s) in a Subscription, Contributor or higher privileges on the Front Door CDNs are required." -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "[Step 2 of 4] Preparing to fetch all Front Door CDN Endpoints"
+    Write-Host $([Constants]::SingleDashLine)
+
+     # list to store Front Door End Points 
+    $frontDoors = @()
+    $frontDoorEndPoints = @()
+    
+     # To keep track of remediated and skipped resources	
+    $logRemediatedResources = @()	
+    $logSkippedResources=@()	
+
+    # Control Id	
+    $controlIds = "Azure_ApplicationGateway_NetSec_Enable_WAF_Configuration"
+
+    if ([String]::IsNullOrWhiteSpace($FilePath))	
+    {
+        Write-Host "Fetch all Front Door(s) in Subscription: [$($context.Subscription.SubscriptionId)]" -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+
+         # Get all Front Door(s) in a Subscription
+        $frontDoors =  Get-AzFrontDoorCdnProfile -ErrorAction Stop
+
+        # Seperating required properties
+        $totalfrontDoors = ($frontDoors | Measure-Object).Count
+        
+        if($totalfrontDoors -gt 0)
+        {
+            $frontDoors | ForEach-Object {
+                $frontDoor = $_
+                $frontDoorId = $_.Id
+                $resourceGroupName = $_.ResourceGroupName
+                $frontDoorName = $_.Name
+
+                # Get all Frontendpoint(s) for this Front Door.
+                $frontendpoints = ( Get-AzFrontDoorCdnEndpoint -ResourceGroupName $resourceGroupName -ProfileName $frontDoorName -ErrorAction SilentlyContinue) 
```

Comment:
If there is failure in these commands, would we be able to identify difference between no endpoints and failure case?

Reply:
I have included this change thru if else for now.